force-app/main/default/classes/DMLOperations.cls

```diff
@@ -180,15 +216,16 @@ public with sharing class DMLOperations {
 	 * @param contactsToUpsert List of Contacts to be upserted and linked to accounts.
 	 */
 	public static void upsertAccountsWithContacts(List<Contact> contactsToUpsert) {
-		// Loop through each Contact
-
-		// Extract the Account Name from Contact's LastName
+        for(Contact contactIteminLoop: ContactsToUpsert){ // Loop through each Contact
+            Account varAccount = upsertAccount(contactIteminLoop.LastName); // Call the method from 9 to upsert on the basis of Account Name.
```

Comment:
We will let this slide since the directions call for it, but can you figure work out to avoid calling this in a loop? Maybe refactor the method from question 9 to process a list of potential names instead?